compiler/rustc_lint/src/unused.rs

```diff
@@ -865,7 +865,9 @@ trait UnusedDelimLint {
                 (iter, UnusedDelimsCtx::ForIterExpr, true, None, Some(body.span.lo()), true)
             }
 
-            Match(ref head, ..) if Self::LINT_EXPR_IN_PATTERN_MATCHING_CTX => {
+            Match(ref head, _, ast::MatchKind::Prefix)
```

Comment:
I was about to say that is too coarse-grained since it leads to us no longer flagging `(0).match {}` or `(f()).match {}` but after a quick check I realized that we currently don't flag `(f()).await`, `(0).f()` and `(f()).g()` either.

I'm gonna add an item to my TODO list to open an issue for all of these cases unless there already exists one.

Reply:
Yeah, I'd prefer if those were implemented in a principled way. This PR doesn't add more inconsistency.